packages/platform/atoms/destination-calendar/DestinationCalendarSelector.tsx

```diff
@@ -0,0 +1,145 @@
+import { useState, useEffect } from "react";
+
+import { SingleValueComponent } from "@calcom/features/calendars/DestinationCalendarSelector";
+import { OptionComponent } from "@calcom/features/calendars/DestinationCalendarSelector";
+import { classNames } from "@calcom/lib";
+import { useLocale } from "@calcom/lib/hooks/useLocale";
+import type { ConnectedDestinationCalendars } from "@calcom/platform-libraries";
+import { Badge, Select } from "@calcom/ui";
+
+import { getPlaceholderContent } from "../lib/getPlaceholderContent";
+
+export type DestinationCalendarProps = {
+  connectedCalendars: ConnectedDestinationCalendars["connectedCalendars"];
+  destinationCalendar: ConnectedDestinationCalendars["destinationCalendar"];
+  onChange: (value: { externalId: string; integration: string }) => void;
+  isPending?: boolean;
+  hidePlaceholder?: boolean;
+  value: string | undefined;
+  maxWidth?: number;
+  hideAdvancedText?: boolean;
+};
+
+export const DestinationCalendarSelector = ({
+  connectedCalendars,
+  destinationCalendar,
+  onChange,
+  isPending,
+  value,
+  hidePlaceholder,
+  hideAdvancedText,
+  maxWidth,
+}: DestinationCalendarProps): JSX.Element | null => {
+  const { t } = useLocale();
+  const [selectedOption, setSelectedOption] = useState<{
+    value: string;
+    label: string;
+    subtitle: string;
+  } | null>(null);
+
+  useEffect(() => {
+    const selected = connectedCalendars
+      .map((connected) => connected.calendars ?? [])
+      .flat()
+      .find((cal) => cal.externalId === value);
+
+    if (selected) {
+      const selectedIntegration = connectedCalendars.find((integration) =>
+        integration.calendars?.some((calendar) => calendar.externalId === selected.externalId)
+      );
+
+      setSelectedOption({
+        value: `${selected.integration}:${selected.externalId}`,
+        label: `${selected.name} ` || "",
+        subtitle: `(${selectedIntegration?.integration.title?.replace(/calendar/i, "")} - ${
+          selectedIntegration?.primary?.name
+        })`,
+      });
+    }
+  }, [connectedCalendars]);
+
+  const options =
+    connectedCalendars.map((selectedCalendar) => ({
+      key: selectedCalendar.credentialId,
+      label: `${selectedCalendar.integration.title?.replace(/calendar/i, "")} (${
+        selectedCalendar.primary?.integration === "office365_calendar"
+          ? selectedCalendar.primary?.email
```

Comment:
Would be good to have a comment here - why do we use email if its office 365 calendar and if not name?

Reply:
Tbh I don't really know they do this, its just the way they do it in the web app 😅